fcrepo-clients/src/test/java/edu/unc/lib/dl/fcrepo4/RepositoryObjectFactoryTest.java

```diff
@@ -0,0 +1,202 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.fcrepo4;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.rdf.model.Resource;
+import org.fcrepo.client.FcrepoClient;
+import org.fcrepo.client.FcrepoOperationFailedException;
+import org.fcrepo.client.FcrepoResponse;
+import org.fcrepo.client.PostBuilder;
+import org.fcrepo.client.PutBuilder;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.rdf.Premis;
+import edu.unc.lib.dl.test.SelfReturningAnswer;
+
+/**
+ *
+ * @author bbpennel
+ * @author harring
+ *
+ */
+public class RepositoryObjectFactoryTest {
+
+    @Mock
+    private LdpContainerFactory ldpFactory;
+    @Mock
+    private RepositoryObjectCacheLoader objectCacheLoader;
+    @Mock
+    private RepositoryObjectDataLoader dataLoader;
+    @Mock
+    private FcrepoClient fcrepoClient;
+    @Mock
+    private PutBuilder mockPutBuilder;
+    @Mock
+    private PostBuilder mockPostBuilder;
+    @Mock
+    private FcrepoResponse mockResponse;
+    @Mock
+    private PID pid;
+
+    private RepositoryObjectFactory repoObjFactory;
+    private RepositoryPIDMinter pidMinter;
+    private List<URI> linkHeaders;
+
+    @Before
+    public void init() throws FcrepoOperationFailedException, URISyntaxException {
+        initMocks(this);
+        repoObjFactory = new RepositoryObjectFactory();
+        repoObjFactory.setClient(fcrepoClient);
+        repoObjFactory.setLdpFactory(ldpFactory);
+        pidMinter = new RepositoryPIDMinter();
+        repoObjFactory.setPidMinter(pidMinter);
+        linkHeaders = new ArrayList<>();
+        URI testHeader = new URI("/path/to/resource");
+        linkHeaders.add(testHeader);
+
+        mockPutBuilder = mock(PutBuilder.class, new SelfReturningAnswer());
+        when(fcrepoClient.put(any(URI.class))).thenReturn(mockPutBuilder);
+        when(mockPutBuilder.perform()).thenReturn(mockResponse);
+
+        mockPostBuilder = mock(PostBuilder.class, new SelfReturningAnswer());
+        when(fcrepoClient.post(any(URI.class))).thenReturn(mockPostBuilder);
+        when(mockPostBuilder.perform()).thenReturn(mockResponse);
+        when(mockResponse.getLinkHeaders(any(String.class))).thenReturn(linkHeaders);
+
+    }
+
+    @Test
+    public void mintDepositRecordPidTest() {
```

Comment:
This method actually belongs in a PidMinter unit test

Reply:
created a new test class for PIDMinter